unified-codes/apps/web/src/components/containers/explorer/ExplorerTablePagination.tsx

```diff
@@ -1,52 +1,98 @@
 import { ExplorerSelectors } from '../../../selectors';
-import { IState } from '../../../types';
+import { IExplorerParameters, IState } from '../../../types';
 import { ITheme } from '../../../styles';
 
 const styles = (theme: ITheme) => ({
-    root: { background: theme.palette.background.toolbar, bottom: 0, position: 'sticky' as Position }
+  root: { background: theme.palette.background.toolbar, bottom: 0, position: 'sticky' as Position },
 });
 
-export interface ExplorerTablePaginationProps extends Omit<TablePaginationProps, 'classes'> {
-    classes?: {
-        root?: string;
-        pagination?: string;
-    }
+export interface ExplorerTablePaginationProps
+  extends Omit<TablePaginationProps, 'classes' | 'onChangePage' | 'onChangeRowsPerPage'> {
+  classes?: {
+    root?: string;
+    pagination?: string;
+  };
+  onChangePage?: (page: number, parameters?: IExplorerParameters) => void;
+  onChangeRowsPerPage?: (rowsPerPage: number, parameters?: IExplorerParameters) => void;
+  parameters?: IExplorerParameters;
 }
 
 export type ExplorerTablePagination = React.FunctionComponent<ExplorerTablePaginationProps>;
 
-export const ExplorerTablePaginationComponent: ExplorerTablePagination = ({ classes, ...props }) => (
+export const ExplorerTablePaginationComponent: ExplorerTablePagination = ({
+  classes,
+  parameters,
+  onChangePage,
+  onChangeRowsPerPage,
+  ...props
+}) => {
+  const handleChangePage = (_: React.MouseEvent<HTMLButtonElement> | null, page: number) => {
+    onChangePage && onChangePage(page, parameters);
+  };
+
+  const handleChangeRowsPerPage = (
+    event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
+  ) => {
+    const rowsPerPage = +event.target.value;
+    onChangeRowsPerPage && onChangeRowsPerPage(rowsPerPage, parameters);
+  };
+
+  return (
     <TableFooter classes={{ root: classes?.root }}>
-        <TableRow>
-            <TablePagination classes={{ root: classes?.pagination }} {...props}/>
-        </TableRow>
+      <TableRow>
+        <TablePagination
+          classes={{ root: classes?.pagination }}
+          {...props}
+          onChangePage={handleChangePage}
+          onChangeRowsPerPage={handleChangeRowsPerPage}
+        />
+      </TableRow>
     </TableFooter>
-);
+  );
+};
 
 const mapDispatchToProps = (dispatch: React.Dispatch<IExplorerAction>) => {
-    const onChangePage = (_: React.MouseEvent<HTMLButtonElement> | null, page: number) => dispatch(ExplorerActions.updatePage(page));
-    const onChangeRowsPerPage = (event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => dispatch(ExplorerActions.updateRowsPerPage(+event.target.value));
-    
-    return { onChangePage, onChangeRowsPerPage };
+  const onChangePage = (page: number, parameters: IExplorerParameters) =>
+    batch(() => {
+      dispatch(ExplorerActions.updatePage(page));
+      dispatch(ExplorerActions.updateEntities({ ...parameters, page }));
```

Comment:
As above, I think best to avoid this sort of thing. I don't think it will churn out more performance, and it introduces multiple ways of doing the same thing, which I think is possibly likely to introduce confusion/bugs?

Reply:
Also, ideally we want the source-of-truth to always be "up-to-date". Giving the saga access to newer data before the store is updated feels wrong to me (just me?).

Reply:
Also, does this behave differently when the parameters are removed? I think `batch` only effects render cycles, I think the dispatches are still performed sequentially?

Reply:
true, re the `batch` - this only affects render cycles and isn't helpful here.
Dispatches are themselves synchronous, but I understood that the store isn't necessarily synchronously updated. I think though, that I might be confusing React behaviour - react will asynchronously batch updates to components, so the state which a component sees isn't updated directly after an action is dispatched. 

I'll rework and see if it hangs together. I'm concerned that if it doesn't run synchronously then that particular bug will be difficult to discover.